Source/WebCore/inspector/agents/page/PageTimelineAgent.cpp

```diff
@@ -202,7 +202,15 @@ void PageTimelineAgent::didLayout(RenderObject& root)
     ASSERT(quads.size() >= 1);
     if (quads.size() >= 1)
         TimelineRecordFactory::appendLayoutRoot(entry->data.get(), quads[0]);
+}
 
+void PageTimelineAgent::didLayout()
+{
+    auto* entry = lastRecordEntry();
+    if (!entry)
+        return;
+
+    ASSERT(entry->type == TimelineRecordType::Layout);
```

Comment:
btw none of these lines are really needed as `InspectorTimelineAgent::didCompleteCurrentRecord` already performs these checks